src/embed/mod.rs

```diff
@@ -0,0 +1,138 @@
+mod ffi;
+
+use crate::boxed::ZBox;
+use crate::embed::ffi::ext_php_rs_embed_callback;
+use crate::ffi::{
+    _zend_file_handle__bindgen_ty_1, php_execute_script, zend_eval_string, zend_file_handle,
+    zend_stream_init_filename, ZEND_RESULT_CODE_SUCCESS,
+};
+use crate::types::{ZendObject, Zval};
+use crate::zend::ExecutorGlobals;
+use parking_lot::{const_rwlock, RwLock};
+use std::ffi::{c_char, c_void, CString};
+use std::path::Path;
+use std::ptr::null_mut;
+
+pub struct Embed;
+
+#[derive(Debug)]
+pub enum EmbedError {
+    InitError,
+    ExecuteError(Option<ZBox<ZendObject>>),
+}
+
+static RUN_FN_LOCK: RwLock<()> = const_rwlock(());
+
+impl Embed {
+    pub fn run_script<P: AsRef<Path>>(path: P) -> bool {
+        let path = CString::new(path.as_ref().to_str().unwrap()).unwrap();
+        let mut file_handle = zend_file_handle {
+            handle: _zend_file_handle__bindgen_ty_1 { fp: null_mut() },
+            filename: null_mut(),
+            opened_path: null_mut(),
+            type_: 0,
```

Comment:
I think you can rely on `zend_stream_type` here

Reply:
I don't really care here about the value, it's correctly initialized when using the `zend_stream_init_filename` (in php code it initialize an empty structure without values)